onecodex/analyses.py

```diff
@@ -140,9 +150,12 @@ def _metadata_fetch(self, metadata_fields, label=None, match_taxonomy=True):
             if isinstance(f, tuple):
                 # joined categorical metadata
                 for field in f:
+                    if not isinstance(field, str):
+                        raise OneCodexException(f"Metadata field {field} must be of type str")
```

Comment:
Should we enforce strings, or could we instead stringify ints/floats?

Reply:
This is enforcing the *field name* is a string -- I'll update the error message to make that clearer!

Reply:
(string field names has always been a requirement of `_metadata_fetch()`, it would just throw a hard to interpret error previously)